test/unit/components/studio-comments.test.jsx

```diff
@@ -1,7 +1,13 @@
 import React from 'react';
 import {mountWithIntl} from '../../helpers/intl-helpers.jsx';
+
 import {StudioComments} from '../../../src/views/studio/studio-comments.jsx';
 
+// Replace customized studio comment with default comment to avoid redux issues in the test
```

Comment:
This seems a little dangerous to me if we end up having project comments and studio comments diverge in more ways.

Reply:
Yeah its not ideal. Speaking of unit testing difficulties, @picklesrus one thing that is tough is unit testing components that require the redux provider in the chain. There are a couple examples of using mock-store, but I had difficulty doing that because then i couldn't manually call `setProps` on the component (it needs to be called on the top level element, but wrapping it in a Provider makes it not the top level element). I could try instead mocking the StudioComment component and wrapping that with a provider. Alternatively, i could try the shallow render method that would only render `StudioComments` and not render all the children.

Reply:
I'd also like to look into how to move the logic that interacts with data fetching (useEffect, isAdmin state/ref) into a custom hook to encapsulate it and then test that independently of testing the component stuff like which things get rendered.

Reply:
I haven't looked at the code, but could you use dive to get into the component you want and set its props?

Reply:
I took another look at the tests. It looks like I can avoid needing to figure out how to provide a redux store by using `shallowWithIntl` instead of `mountWithIntl` because it only renders on level deep, so the comment component (nested within TopLevelComment) never needs to be rendered. This is good because it wasn't part of any of these tests anyway, so its probably good to only render as much as you need to accomplish the tests.

I also took a step to disaggregating the callback logic tests from the render tests, by putting them in different places. The callback logic tests incidentally still need to use `mountWithIntl` because annoyingly even though `mount` and `shallow` try to provide a consistent component wrapper (ie you can call setProps and update with both), `shallow` will not run calllbacks in `useEffect`.

Reply:
@picklesrus unfortunately i can't call `setProps` on anything that isn't the top level element, so although i can get a reference to the component using `dive` or `find`, it isn't a drop in fix because i can't call `setProps` on it...